Xamarin.Forms.Core/Page.cs

```diff
@@ -317,6 +317,8 @@ public void SendAppearing()
 
 			var pageContainer = this as IPageContainer<Page>;
 			pageContainer?.CurrentPage?.SendAppearing();
+
+			Application.Current?.OnPageAppearing(this);
```

Comment:
it shouldn't be raised on the `Application.Current`, but on the actual `Application` instance on which the page appears (or disappears)

Reply:
I don't see a `this.Application` or the like. What's the API I should be using instead?

Reply:
Also, since new App() [sets it as current](https://github.com/xamarin/Xamarin.Forms/blob/master/Xamarin.Forms.Core/Application.cs#L32) I wonder what's the problem of doing it this way?

Is it even supported to have multiple `Application` instances? None of our templates suggest so...